src/Pagination/Connection.php

```
@@ -0,0 +1,293 @@
+        ]);
+    }
+
+    /**
+     * Returns the collection resolved with the pageInfo provided.
+     *
+     * @param mixed $value
+     * @param array $args
+     * @param array $context
+     * @param ResolveInfo $info
+     *
+     * @return array
+     */
+    public function resolve($value, $args, $context, ResolveInfo $info)
+    {
+        $result = call_user_func_array(
+            $this->connectionResolver,
+            func_get_args()
+        );
+
+        if(!$result instanceof SS_List) {
+            throw new \Exception('Connection::resolve() must resolve to a SS_List instance.');
+        }
+
+        return $this->resolveList($result, $args, $context, $info);
+    }
+
+    /**
+     * Wraps an {@link SS_List} with the required data in order to return it as
+     * a response. If you wish to resolve a standard array as a list use
+     * {@link ArrayList}.
+     *
+     * @param SS_List $list
+     *
+     * @return array
+     */
+    public function resolveList($list, $args, $context = null, $info = null) {
+        $limit = (isset($args['limit']) && $args['limit']) ? $args['limit'] : $this->defaultLimit;
+        $offset = (isset($args['offset'])) ? $args['offset'] : 0;
+
+        if($limit > $this->maximumLimit) {
+            $limit = $this->maximumLimit;
+        }
+
+        $nextPage = false;
+        $previousPage = false;
+        $count = $list->count();
+
+        if($list instanceof Limitable) {
+            $list = $list->limit($limit, $offset);
+
+            if($limit && (($limit + $offset) < $count)) {
+                $nextPage = true;
+            }
+
+            if($offset > 0) {
+                $previousPage = true;
+            }
+        }
+
+        if($list instanceof Sortable) {
+            if(isset($args['sort'])) {
+                $direction = (isset($args['sortDirection'])) ? $args['sortDirection'] : 'ASC';
+
+                if(in_array($args['sort'], $this->sortableFields)) {
```

If you're sorting by a field that's not whitelisted, it should throw an exception (which will automatically surface in the GraphQL query response)